src/devices/bus/isa/ibm_speech.cpp

```diff
@@ -0,0 +1,497 @@
+// license:BSD-3-Clause
+// copyright-holders:Katherine Rohl
+/*
+    IBM PS/2 Speech Adapter
+
+    The ROM listing calls this card the "TALKER II".
+
+    The ISA version of the card uses a different timing mechanism
+    than the PCjr version to account for a 4MHz operating speed
+    rather than the 4.77 bus clock. The ISA version also uses
+    IRQ7 rather than IRQ1.
+
+    Lots of TTL, all off-the-shelf parts except the mask ROM.
+
+    Y1      - 4MHz oscillator
+    ZM5     - MC14529CP
+    ZM8     - TMS5220CNL
+    ZM9     - 8254 PIT
+    ZM10    - 32Kx8 mask ROM
+    ZM22    - 8255A PIC
+*/
+
+#include "emu.h"
+#include "xtal.h"
+#include "ibm_speech.h"
+
+#define LOG_ROM         (1 << 0)
+#define LOG_MUX         (1 << 1)
+#define LOG_CVSD        (1 << 2)
+#define LOG_PIT         (1 << 3)
+#define LOG_IRQ         (1 << 4)
+#define LOG_PORTS       (1 << 5)
+#define LOG_ALL         (LOG_ROM|LOG_MUX|LOG_CVSD|LOG_PIT|LOG_IRQ|LOG_PORTS)
+
+//#define VERBOSE         (LOG_PORTS)
+#include "logmacro.h"
+
+#ifdef _MSC_VER
+#define FUNCNAME __func__
+#else
+#define FUNCNAME __PRETTY_FUNCTION__
+#endif
+
+#define XTAL_Y1     XTAL(4'000'000)
+
+DEFINE_DEVICE_TYPE(ISA8_IBM_SPEECH, isa8_ibm_speech_device, "isa_ibm_speech", "IBM PS/2 Speech Adapter")
+
+//-------------------------------------------------
+//  constructor
+//-------------------------------------------------
+
+isa8_ibm_speech_device::isa8_ibm_speech_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock) :
+	device_t(mconfig, ISA8_IBM_SPEECH, tag, owner, clock),
+	device_isa8_card_interface(mconfig, *this),
+	m_lpc(*this, "lpc"),
+	m_cvsd(*this, "cvsd"),
+	m_8254_audio(*this, "8254_audio"),
```

Comment:
I'd rename this to something more explicit as `m_speaker` / `m_beeper` / `m_dac1bit`, 8254 is the PIT timer, not necessarily the actual output. Same deal for `m_timers` just below, I'd call it `m_pit` instead.